docs/fundamentals/flow/monitoring-flow.md

```diff
@@ -0,0 +1,140 @@
+# Monitor Flow
+
+```{caution} 
+The monitoring feature is still in Beta and the API is not stable yet.
+```
+
+A Jina {ref}`Flow <flow-cookbook>` exposes several core metrics that allow you to have a deeper look
+on what is happening inside it. Metrics are particularly useful for building dashboards and alerts, with them, you can, for example, monitor the overall performance 
+of your Flow, detect bottlenecks, alert your team when some component of your Flow is down.
+
+Jina Flow expose metrics in the [prometheus format](https://prometheus.io/docs/instrumenting/exposition_formats/) which
+is plain text that is both understandable by human and machine. These metrics are intended to be scraped by
+[prometheus](https://prometheus.io/) which is an industry standard for monitoring. 
+We recommend you to visualize metrics with [grafana](https://grafana.com/)
+
+
+## Using the monitoring in a Flow
+
+A {ref}`Flow <flow-cookbook>` is composed of several Pods namely the Gateway and the Executors (potentially a Head as well). 
+see the {ref}`architecture overview <architecture-overview>` for more details. Each of these Pods are microservices which communicate 
+by gRPC. They will expose their own metrics using the [prometheus client](https://prometheus.io/docs/instrumenting/clientlibs/).
+It means that they are as many metrics endpoints that they are Pods in your Flow. 
+
+Let's give an example to illustrate it :
+
+````{tab} via Python API
+
+This example shows how to start a Flow with monitoring enabled via the Python API:
+
+```python
+from jina import Flow
+
+with Flow(monitoring=True, port_monitoring=9090).add(
+    uses='jinahub://SimpleIndexer', port_monitoring=9091
+) as f:
+    f.block()
+```
+````
+
+````{tab} via YAML
+This example shows how to start a Flow with monitoring enabled via yaml:
+
+In a `flow.yaml` file
+```yaml
+!Flow
+with:
+  monitoring: true
+  port_monitoring: 9090
+executors:
+- uses: jinahub://SimpleIndexer
+  port_monitoring: 9091
+```
+
+```bash
+jina flow --uses flow.yaml
+```
+````
+
+This Flow will create two Pods, one for the Gateway, one for the SimpleIndexer Executor, and therefore it will create two 
+metrics endpoints:
+
+* `http://localhost:9090  ` for the gateway
+* `http://localhost:9091  ` for the SimpleIndexer
+
+````{admonition} Default Monitoring port
+:class: hint
+The default monitoring port is `9090`, if you want to enable the monitoring on both the Gateway and the Executors you need to specifiy
+the `prometheus_port` for the Executors. 
+````
+
+
+Because each Pods in a Flow expose its own metrics the monitoring feature could be used independently on each Pod.
+It means that you are not forced to always monitor each Pods of your Flow. For example, you could be only interested in
+metrics coming from the Gateway, and therefore you only activate the monitoring on it. On the other hand you might be only
+interested in monitoring a single Executor. Note that by default the monitoring is disabled everywhere.
```

Comment:
```suggestion
Because each Pods in a Flow exposes its own metrics, the monitoring feature can be used independently on each Pod.
This means that you are not forced to always monitor every Pods of your Flow. For example, you could be only interested in
metrics coming from the Gateway, and therefore you only activate the monitoring on it. On the other hand, you might be only
interested in monitoring a single Executor. Note that by default the monitoring is disabled everywhere.
```